app/controllers/events_controller.rb

```diff
@@ -0,0 +1,32 @@
+class EventsController < ApplicationController
+  before_action :authenticate_user!, except: %i[index]
+
+  def index
+    @future_events = Event.all.upcoming
+    @past_events = Event.all.past
```

Comment:
Refactor this code: Currently, the implementation seems difficult to understand. I believe it should be similar to this:
@future_events = Event.upcoming
@past_events = Event.past